### PR TITLE
add e2e tests for AppWrappers with Jobs, Deployments, and Services

### PR DESCRIPTION
**Pull request by dgrove-oss**
Follow up from #640  -- added two additional e2e tests for AppWrappers.  The Deployment+Service one would have failed before the fix.

### CODE REVIEW

File: test/e2e/deployment_appwrapper_test.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	"sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+func TestDeploymentAppWrapperCpu(t *testing.T) {
+	runDeploymentAppWrapper(t)
+}
+
+func TestDeploymentAppWrapperGpu(t *testing.T) {
+	runDeploymentAppWrapper(t)
+}
```
Comment (sutaakar) [inline test/e2e/deployment_appwrapper_test.go:42]: Just a note, in other tests the CPU/GPU test split is done as GPU test has adjusted configuration.
If you don't have any specific GPU related code feel free to keep only one test function.

Comment (dgrove-oss) [reply test/e2e/deployment_appwrapper_test.go]: Thanks.  Done.

**Comment by dgrove-oss** [comment]
Test cleaned up and ready for review.
On my cluster with the codeflare operator from RHOAI 2.16 installed `TestDeploymentAppWrapper` fails (as expected because it doesn't have the fix from #640).

**Review by varshaprasad96 — APPROVED** [review]
/lgtm

**Pull request merged**